open-apparel-registry/open-apparel-registry
# Require authentication for facility CSV downloads and log requests

## Overview

Require a login to download a facility CSV from the map page and log all CSV download requests to the database. 

Connects #668

## Demo

<img width="841" alt="Screen Shot 2019-07-23 at 7 10 43 PM" src="https://user-images.githubusercontent.com/17363/61759568-9c852700-ad7d-11e9-8cf8-69ad56a35b94.png">

## Notes

We have kept the method, model, and model attribute names relatively generic so we have the option of logging other downloads using the same code while avoiding the need to change names.

## Testing Instructions

This assumes the fixture data has been loaded and processed with `./scripts/resetdb`

* Log in as `c2@example.com`
* Search for `shoes`
* Click "Download CSV" and verify that is behaves as before.
* Run `./scripts/manage dbshell`
* Run `select * from api_downloadlog` and verify that the `path` and `record_count` columns match.
* Stop `./scripts/server`. Click "Download CSV" and verify that error toast is shown. Start `./scripts/server`
* Log out
* Search for "shoes" again and click "Download CSV." Verify that the "Log In To Download" dialog is shown.
  * Verify each button behavior (Cancel, Register, and Log In). 

## Checklist
